### PR TITLE
Add a -Chord param to Get-PSReadLineKeyHandler

### LINKED ISSUE #879: Add a -Key parameter to Get-PSReadLineKeyHandler
Instead of:

```powershell
Get-PSReadLineKeyHandler | ? Key -eq Enter
```

it would be nice to be able to do:

```powershell
Get-PSReadLineKeyHandler -Key Enter
```

both for convenience and for symmetry with `Set-PSReadLineKeyHandler`

Environment data
----------------

```powershell
PSReadline version: 2.0.0-beta3
```

### PR DESCRIPTION
**Pull request by theaquamarine**
Allow filtering of results by keys used in binding.

Closes #879

### CODE REVIEW

File: PSReadLine/Options.cs [hunk 1]
```diff
@@ -271,24 +271,41 @@ public static void RemoveKeyHandler(string[] key)
...
+        /// Helper function for the Get-PSReadLineKeyHandler cmdlet.
+        /// </summary>
+        /// <returns></returns>
+        public static IEnumerable<PowerShell.KeyHandler> GetKeyHandlers(bool includeBound, bool includeUnbound, string[] Chord)
         {
             var boundFunctions = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
 
+            var searchChords = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+            if (Chord != null) { foreach (var Key in Chord) { searchChords.Add(Key); }; }
```
Comment (daxian-dbw) [inline PSReadLine/Options.cs]: This means you don't different keys like "Ctrl+k" from "Ctrl+K", but they are different key bindings.
For "Ctrl+k", it's triggered by pressing <kbd>Ctrl</kbd> and <kbd>k</kbd>
For "Ctrl+K", it's triggered by pressing <kbd>Ctrl</kbd>, <kbd>Shift</kbd>, and <kbd>k</kbd>

To accurately find the key, you will need to do the similar to [this code](https://github.com/PowerShell/PSReadLine/blob/03a0d3fe2b715eee66dcdc1304cac9e03456b434/PSReadLine/Options.cs#L178-L179), to get the `PSKeyInfo` that the specified string chord represents.

File: PSReadLine/Cmdlets.cs [hunk 2]
```diff
@@ -825,6 +825,10 @@ public SwitchParameter Unbound
         }
         private SwitchParameter? _unbound;
 
+        [Parameter]
+        [Alias("Key")]
+        public string[] Chord { get; set; }
```
Comment (daxian-dbw) [inline PSReadLine/Cmdlets.cs:832]: This should be put in a separate parameter set. When specifying `-Chord`, it implies searching from the bound bindings.
After adding this parameter, the syntax should look like below, and the `[-Bound] [-Unbound]` parameter set should be made the default.
```
PS:82> gcm Get-PSReadLineKeyHandler -Syntax

Get-PSReadLineKeyHandler [-Bound] [-Unbound] [<CommonParameters>]

Get-PSReadLineKeyHandler -Chord [<CommonParameters>]
```

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: PSReadLine/Cmdlets.cs (hunk 2 above)
Comment (theaquamarine) [reply PSReadLine/Cmdlets.cs:832]: I've added it to a separate ParameterSet, made it mandatory for that ParameterSet, and allowed it to be positional.

File: PSReadLine/Options.cs (hunk 1 above)
Comment (theaquamarine) [reply PSReadLine/Options.cs]: Implemented

File: PSReadLine/Options.cs [hunk 3]
```diff
@@ -271,17 +271,64 @@ public static void RemoveKeyHandler(string[] key)
...
+                                {
+                                    Key = firstKey.KeyStr + "," + secondKey.KeyStr,
+                                    Function = entry.BriefDescription,
+                                    Description = entry.LongDescription,
+                                    Group = GetDisplayGrouping(entry.BriefDescription),
+                                };
+                            }
+                        }
+                    }
+                }
```
Comment (daxian-dbw) [inline PSReadLine/Options.cs]: Key bindings in VI mode are not covered by this. Can you please add the support to VI mode too?

File: PSReadLine/Cmdlets.cs [hunk 4]
```diff
@@ -809,22 +809,27 @@ public object GetDynamicParameters()
     [OutputType(typeof(KeyHandler))]
```
Comment (daxian-dbw) [inline PSReadLine/Cmdlets.cs:810]: Can you please add `DefaultParameterSetName = "FullListing"` to the `Cmdlet(...)` attribute, as follows:
```c#
[Cmdlet("Get", "PSReadLineKeyHandler", DefaultParameterSetName = "FullListing", HelpUri = "https://go.microsoft.com/fwlink/?LinkId=528807")]
```

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: PSReadLine/Cmdlets.cs [hunk 5]
```diff
@@ -809,22 +809,27 @@ public object GetDynamicParameters()
...
+        [Parameter(ParameterSetName = "FullListing")]
         public SwitchParameter Unbound
         {
             get => _unbound.GetValueOrDefault();
             set => _unbound = value;
         }
         private SwitchParameter? _unbound;
 
+        [Parameter(ParameterSetName = "SpecificBindings", Position = 0, Mandatory = true)]
+        [ValidateNotNullOrEmpty]
```
[the remark this answers is not in the document]
Comment (msftrncs) [reply PSReadLine/Cmdlets.cs:830]: Does `Mandatory` also enforce `NotEmpty` or give a clear error message?  I have less experience in C# cmdlet writing, but in PowerShell, using the `Mandatory` property doesn't get quite the same affect as `ValidateNotNullOrEmpty()`

[the remark this answers is not in the document]
Comment (daxian-dbw) [reply PSReadLine/Cmdlets.cs:830]: Good point. The parameter type is a string array, so `ValidateNotNullOrEmpty` does more than the default check from `mandatory`. This attribute should be kept.

**Review by daxian-dbw — APPROVED** [review]
LGTM!
Sorry for the long delay, and thanks a lot for your contribution!

**Pull request merged**